enterprise/cloud.oracle/src/org/netbeans/modules/cloud/oracle/assets/Steps.java

```diff
@@ -92,16 +60,20 @@ public static synchronized Steps getDefault() {
         return instance;
     }
 
-    public CompletableFuture<Object> executeMultistep(Step firstStep, Lookup lookup) {
+    public CompletableFuture<Values> executeMultistep(AbstractStep firstStep, Lookup lookup) {
         DialogDisplayer dd = DialogDisplayer.getDefault();
         CompletableFuture future = new CompletableFuture();
         RP.post(() -> {
-            Multistep multistep = new Multistep(firstStep, lookup);
-            NotifyDescriptor.ComposedInput ci = new NotifyDescriptor.ComposedInput(Bundle.AddSuggestedItem(), 3, multistep.createInput());
-            if (DialogDescriptor.OK_OPTION == dd.notify(ci)) {
-                future.complete(multistep.getResult());
-            } else {
-                future.cancel(true);
+            try {
+                Multistep multistep = new Multistep(firstStep, lookup);
+                NotifyDescriptor.ComposedInput ci = new NotifyDescriptor.ComposedInput(Bundle.AddSuggestedItem(), 3, multistep.createInput());
+                if (DialogDescriptor.OK_OPTION == dd.notify(ci)) {
```

Comment:
I'd use `dd.notifyFuture(ci).handle()` or `.then*` so that  RP (throughput 1) is not blocked.